app/components/autocomplete/slash_suggestion/slash_suggestion.js

```diff
@@ -0,0 +1,208 @@
+                intl.formatMessage({
+                    id: 'mobile.commands.error_message',
+                    defaultMessage: 'There was an error executing your command. To send a message beginning with "/", try adding an empty space at the beginning of the message.'
+                }));
+            return;
+        }
+
+        if (currentTeamId !== nextTeamId) {
+            this.setState({
+                lastCommandRequest: 0
+            });
+        }
+
+        const regex = SLASH_REGEX;
+        const match = nextValue.match(regex);
+
+        if (!match || this.state.suggestionComplete) {
+            this.setState({
+                active: false,
+                matchTerm: null,
+                suggestionComplete: false
+            });
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        const dataIsStale = Date.now() - this.state.lastCommandRequest > (1000 * 60 * 5); // 5 mins
+
+        if ((!nextCommands.length || dataIsStale) && nextCommandsRequest.status !== RequestStatus.STARTED) {
+            this.props.actions.getAutocompleteCommands(nextProps.currentTeamId);
+            this.setState({
+                lastCommandRequest: Date.now()
+            });
+        }
+
+        const matchTerm = match[2];
+
+        const data = this.filterSlashSuggestions(matchTerm, nextCommands);
+
+        this.setState({
+            active: data.length,
+            dataSource: data
+        });
+
+        this.props.onResultCountChange(data.length);
+    }
+
+    filterSlashSuggestions = (matchTerm, commands) => {
+        return commands.filter((command) => {
+            if (!command.auto_complete) {
+                return false;
+            } else if (!matchTerm) {
+                return true;
+            }
+
+            return command.display_name.startsWith(matchTerm) || command.trigger.startsWith(matchTerm);
+        });
+    }
+
+    completeSuggestion = (command) => {
+        const {onChangeText} = this.props;
+
+        const completedDraft = `/${command} `;
+
+        onChangeText(completedDraft);
+
+        this.setState({
+            active: false,
+            suggestionComplete: true
+        });
+    };
+
+    keyExtractor = (item) => item.id || item.trigger;
+
+    renderItem = ({item}) => {
+        const style = getStyleFromTheme(this.props.theme);
+
+        return (
+            <TouchableOpacity
+                onPress={() => this.completeSuggestion(item.trigger)}
+                style={style.row}
+            >
+                <Text style={style.suggestionName}>{`/${item.display_name || item.trigger} ${item.auto_complete_hint}`}</Text>
+                <Text style={style.suggestionDescription}>{item.auto_complete_desc}</Text>
+            </TouchableOpacity>
+        );
+    };
+
+    getItemLayout = ({index}) => ({length: ROW_HEIGHT, offset: ROW_HEIGHT * index, index})
```

Comment:
this is not being used right? and actually we won't scroll programmatically so there's no need for this here